packages/cli/src/services/project-checker.ts

```diff
@@ -69,3 +71,66 @@ export async function checkResourceExists(name: string, placementDir: string, ex
+  const cliVersion = userAgent.split(' ')[0].split('/')[2]
+  await compareVersionsAndDisplayMessages(cliVersion, projectVersion, projectPath)
+}
+
+async function getBoosterVersion(projectPath: string): Promise<string> {
+  const projectAbsolutePath = path.resolve(projectPath)
+  try {
+    const packageJsonContents = require(path.join(projectAbsolutePath, 'package.json'))
+    const version = packageJsonContents.dependencies['@boostercloud/framework-core']
+    const versionParts = version.replace('^','').replace('.tgz','').split('-')
+    return versionParts[versionParts.length-1]    
+  } catch (e) {
+    throw new Error(
+      `There was an error when recognizing the application. Make sure you are in the root path of a Booster project:\n${e.message}`
+    )
+  }
+}
+
+async function compareVersionsAndDisplayMessages(cliVersion: string, projectVersion: string, projectPath: string): Promise<void> {
+  if (cliVersion === projectVersion)  { return }
+  const cliVersionParts = cliVersion.split('.').map((v) => parseInt(v,10))
+  const projectVersionParts = projectVersion.split('.').map((v) => parseInt(v,10))
+  if (cliVersionParts.length !== projectVersionParts.length) {
+    throw new Error(`Versions must have the same length. CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+  }
+  if (projectVersionParts.length !== 3) {
+    throw new Error(`Versions must follow semantic convention X.Y.Z | CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+  }
+  if (cliVersionParts[0] === projectVersionParts[0]) {
+    if (cliVersionParts[1] === projectVersionParts[1]) {
+      //differences in the 'fix' part
+      if (cliVersionParts[2] !== projectVersionParts[2]) {
+        logger.info(`WARNING: Project Booster version differs in the 'fix' section. CLI version: ${cliVersion}. Project Booster version: ${projectVersion}`)
+      }
+    } else if (cliVersionParts[1] > projectVersionParts[1]) {
+      //cli higher than project in 'feat' section
+      const promptMsg = `@boostercloud/cli version ${cliVersion} is higher than project version (${projectVersion}) in the 'feature' section. Do you want to upgrade your project dependencies?`
+      const value: string = await new Prompter().defaultOrChoose(undefined,promptMsg,['Yes','No'])
+      if (value == 'Yes') {
+        await updatePackageJsonDependencyVersions(cliVersion, projectPath) //update package.json
+        logger.info(`package.json Booster dependencies have been updated to version ^${cliVersion}.`)
+      } else {
+        throw new Error(`CLI version ${cliVersion} is higher than your project Booster version ${projectVersion}. Please upgrade your project dependencies.`)
+      }
+    } else {
+      //cli lower than project in 'feat' section
+      throw new Error(`CLI version ${cliVersion} is lower than your project Booster version ${projectVersion}. Please upgrade your @boostercloud/cli to the same version with npm`)
```

Comment:
I'd do the check explicit here as well:

```ts
if (semverLowerVersion(cliVersion, projectVersion) {
  throw lowerVersionError(...)
}
```

Reply:
I have created custom Error subclasses for these cases when versions differ